### PR TITLE
Carpy the Legal Carp gets a suit for differentiation

### PR DESCRIPTION
**Pull request by ghost**
## About the PR

Adds a RSI for the legal carp that adds a "suit" (big grey rectangle on the sprite) and a "tie" (barely see-able based on how carp coloring works.)
Furthermore, adds functionality for "BaseUnshadedAccessory" as a layer (thanks Timemaster!!!!)
## Why / Balance

Wanted something that makes Carpy visually unique so that the local greytider doesn't kill him immediately because "CARP!!"
## Technical details

RSI + meta.json + one line of YAML
## Media

image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Adeinitas, Timemaster99
- add: Gave Carpy the legal carp a suit or shirt.

### CODE REVIEW

**Comment by ghost** [comment]
Deltanedas you're gonna love this one because it's fish

**Comment by deltanedas** [comment]
the suit is getting multiplied by the random colour which makes it very hard to see, put it in a separate layer (similar to how it uses an unshaded layer for glowing teeth)

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml [hunk 1]
```diff
@@ -63,6 +63,8 @@
   - type: NpcFactionMember
     factions:
     - PetsNT
+  - type: Sprite
+    sprite: DeltaV\Mobs\Pets\lawyercarp.rsi # Fancy Fishe
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml]: noob use /

File: Resources/Textures/DeltaV/Mobs/Pets/lawyercarp.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
+{
+  "version": 1,
+  "size": {
+    "x": 32,
+    "y": 32
+  },
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from https://github.com/tgstation/tgstation/blob/master/icons/mob/animal.dmi, edited 'dead' to create 'crit' and 'dead' states and then edited by Adeinitas on Discord to give it a suit.",
+  "states": [
```
Comment (deltanedas) [inline Resources/Textures/DeltaV/Mobs/Pets/lawyercarp.rsi/meta.json:9]: once you add suit as layers in sprite and mob state visualiser you can probably remove all other states here, layers can specify rsi like this
```yml
layers:
- state: alive
- sprite: DeltaV/Mobs/fishe.rsi
  state: suit
  map: ["whatever"]
```

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml (hunk 1 above)
Comment (ghost) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml]: pro noob moment

File: Resources/Textures/DeltaV/Mobs/Pets/lawyercarp.rsi/meta.json (hunk 2 above)
Comment (ghost) [reply Resources/Textures/DeltaV/Mobs/Pets/lawyercarp.rsi/meta.json:9]: Where do I add the layer in the sprite and mob state visualizer; in fact what IS the sprite and mob state visualizer, I have no idea where to put that, which is why it's gone to the meta.json. Putting a suit on a carp is harder than I thought

Comment (ghost) [reply Resources/Textures/DeltaV/Mobs/Pets/lawyercarp.rsi/meta.json:9]: Wait, nevermind, I'm dumb

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml [hunk 3]
```diff
@@ -63,6 +63,15 @@
   - type: NpcFactionMember
     factions:
     - PetsNT
+  - type: Sprite
+    sprite: DeltaV/Mobs/Pets/lawyercarp.rsi # Fancy Fishe
+    layers:
+      - map: [ "enum.DamageStateVisualLayers.Base" ]
+        state: alive
+      - map: [ "enum.DamageStateVisualLayers.BaseUnshaded" ]
+        state: mouth
```
Comment (DangerRevolution) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml:74]:
```suggestion
       - state: mouth
```

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml [hunk 4]
```diff
@@ -63,6 +63,15 @@
...
+        state: suit
```
Comment (DangerRevolution) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml:77]:
```suggestion
       - state: suit
```

**Review by DangerRevolution — COMMENTED** [review]
I'm on my phone so just check the indentation is correct

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml (hunk 3 above)
Comment (Axionyxx) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml:74]: What are you trying to do here? This isn't how sprite maps work

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/pets.yml:74]: to set shader to unshaded do `shader: unshaded` i dont think that layer exists

**Comment by ghost** [comment]
thanks timemaster 👍

**Comment by ghost** [comment]
my lord, is this funny fishe to your liking?

**Pull request merged**